binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java

```diff
@@ -64,8 +64,13 @@ public void initialize() {
         Configuration config = getThing().getConfiguration();
 
         location = (String) config.get("location");
-        if ("us".equalsIgnoreCase((String) config.get("unit"))) {
+        if ("metric".equalsIgnoreCase((String) config.get("unit"))) {
+            unit = "c";
+        } else if ("us".equalsIgnoreCase((String) config.get("unit"))) {
             unit = "f";
+        } else {
+            // means unit not known
+            unit = null;
```

Comment:
a) the Binding could throw an exception in that case, but then the thing it is not usable. Maybe we can think of the correct behavior when we introduce the additional status and extend the binding again.
b) as we agreed to write the default value into the configuration, there is no need to apply the default again. If the user does not override the value explicitly to null, the configuration will automatically include the default value. So there is no need of default handling inside handlers any more.